src/slurm_plugin/slurm_resources.py

```diff
@@ -231,6 +229,38 @@ def is_poweing_up_idle(self):
     def is_ice(self):
         return self.error_code in self.EC2_ICE_ERROR_CODES
 
+    def is_reboot_requested(self):
+        return self.SLURM_SCONTROL_REBOOT_REQUESTED_STATE in self.states
+
+    def is_reboot_issued(self):
+        return self.SLURM_SCONTROL_REBOOT_ISSUED_STATE in self.states
+
+    def is_rebooting(self):
```

Comment:
The naming of this function is confusing for me.
Reboot nodes with `MIXED/ALLOCATED/IDLE+REBOOT_REQUESTED` will be considered as `is_rebooting` = false, but they are actually rebooting. I understand this doesn't impact the check of `is_state_healthy`. It will return True.

But shall we also include this case in `is_rebooting` function? So in the future even if we change the `is_state_healthy` function, we don't rely on that for whether a node is booting

Reply:
I agree the name is confusing. A clearer (although uglier) name would be `is_rebooting_healthily()`, because in this function we should check if the node is in any rebooting state which might trigger our health checks, and if so return `True` to prevent the node from being replaced.

I will discuss with @demartinofra about the rest of your comments.